src/search/randomdestinationairportpickingbycriteria.h

```diff
@@ -0,0 +1,41 @@
+#ifndef RANDOMDESTINATIONAIRPORTPICKINGBYCRITERIA_H
+#define RANDOMDESTINATIONAIRPORTPICKINGBYCRITERIA_H
+
+#include "search/randomdepartureairportpickingbycriteria.h"
+
+#include <QObject>
+#include <QThread>
+
+namespace atools {
+  namespace geo {
+    class Pos;
+  }
+}
+
+class RandomDestinationAirportPickingByCriteria : public QThread
```

Comment:
This is ok but could be easier implemented using Qt Concurrent https://doc.qt.io/qt-5/qtconcurrent-index.html which is a more high level threading library.